vllm/utils.py

```diff
@@ -2170,6 +2170,11 @@ def _maybe_force_spawn():
     if cuda_is_initialized():
         reason = "CUDA is initialized"
     elif is_in_ray_actor():
+        # even we choose to spawn, we need to pass to the subprocess
+        # the ray address, so that it knows to connect to the ray cluster.
+        # env vars are inherited by subprocesses, even if we use spawn.
+        import ray
+        os.environ["RAY_ADDRESS"] = ray.get_runtime_context().gcs_address
```

Comment:
this is a follow-up after https://github.com/vllm-project/vllm/pull/14705

Reply:
Ah make sense. I only tested the Ray Serve use case before. cc @ruisearch42